--- makefile.g++	2015-03-25 11:07:32.000000000 +0100
+++ makefile_new.g++	2015-12-09 10:56:20.398871080 +0100
@@ -29,11 +29,14 @@
  	tmpltpsr.o tmpltrhs.o tmpltutl.o userdata.o userfunctions.o utility.o watch.o
 
 .c.o :
-	gcc -c -Wall -x c++ -O3 -fno-strict-aliasing -Wundef -Wpointer-arith -Wshadow -Wcast-qual \
+	gcc -c -Wall -x c++ -O3 -fPIC -fno-strict-aliasing -Wundef -Wpointer-arith -Wshadow -Wcast-qual \
 	    -Winline -Wredundant-decls -Waggregate-return -Wno-implicit $< 
 
 clips : $(OBJS)
-	gcc -o clips $(OBJS) -lstdc++   
+	gcc -o clips $(OBJS) -lstdc++ -lm  
+
+libclips.so : $(OBJS)
+	g++ -shared -o libclips.so $(OBJS) -lstdc++ -lm
 
 # Dependencies generated using "gcc -MM *.c"
 
